src/redux/books/books.js

```diff
@@ -1,60 +1,105 @@
 // Imports
-import { v4 as uuidv4 } from 'uuid';
-
-// Action types
-const ADD = 'bookstore/books/ADD_BOOK';
-const REMOVE = 'bookstore/books/REMOVE_BOOK';
-
-// Initial state
-const initialState = [
-  {
-    id: uuidv4(),
-    percent: 64,
-    category: 'Action',
-    title: 'The Hunger Games',
-    author: 'Suzanne Collins',
-    chapter: 'Chapter 17',
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+// Base URL
+const URL = `https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/${process.env.REACT_APP_API_KEY}/books`;
+
+// Thunk for fetching books from API
+export const fetchBooks = createAsyncThunk(
+  'books/getBooks',
+  async () => {
+    const response = await fetch(URL);
+    const json = await response.json();
+    let jsonArr = Object.keys(json).map((item) => {
+      json[item][0].id = item;
+      return json[item][0];
+    });
+    jsonArr = jsonArr.sort((a, b) => {
+      const fa = a.title.toLowerCase();
+      const fb = b.title.toLowerCase();
+
+      if (fa < fb) {
+        return -1;
+      }
+      if (fa > fb) {
+        return 1;
+      }
+      return 0;
+    });
+    return jsonArr;
   },
-  {
-    id: uuidv4(),
-    percent: 30,
-    category: 'Science Fiction',
-    title: 'Dune',
-    author: 'Frank Herbert',
-    chapter: 'Chapter 3: "A Lesson Learned"',
+);
+
+// Thunk for posting books to API
+export const postBook = createAsyncThunk(
+  'books/addBook',
+  async (data) => {
+    await fetch(URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: data,
+    });
   },
-  {
-    id: uuidv4(),
-    percent: 10,
-    category: 'Economy',
-    title: 'Capital in the Twenty-First Century',
-    author: 'Suzanne Collins',
-    chapter: 'Introduction',
+);
+
+// Thunk for deleting a book from API
+export const deleteBook = createAsyncThunk(
+  'books/deleteBook',
+  async (data) => {
+    const data1 = JSON.stringify(data);
+    await fetch(`${URL}/${data.item_id}`, {
+      method: 'DELETE',
+      headers: { 'Content-Type': 'application/json' },
+      body: data1,
+    });
```

Comment:
- [ ] Great work trying to implement the change suggested by the first reviewer. I noticed a bug was introduced when you made the changes. When I delete an item it does not show on the webpage but when I refresh the page, the item that I deleted is fetched and rendered on the page as the first item. This means that the deleted item is persisted. Kindly fix this error 🙏